test/verify-test.ts

```diff
@@ -0,0 +1,240 @@
+import { ValidationError } from 'ripple-api/common/errors'
+import { verifyCommonFields } from './../src/models/transactions/common'
+import { assert } from 'chai'
+
+/**
+ * Transaction Verification Testing
+ *
+ * Providing runtime verification testing for each specific transaction type
+ */
+describe('Transaction Verification', function () {
+    it(`Verifies all optional CommonFields`, () => {
+        const txJson = {
+            Account: "r97KeayHuEsDwyU1yPBVtMLLoQr79QcRFe",
+            TransactionType: "Payment",
+            Fee: "12",
+            Sequence: 100,
+            AccountTxnID: "DEADBEEF",
+            Flags: 15,
+            LastLedgerSequence: 1383,
+            Memos: [
+                {
+                    Memo: {
+                        MemoType: "687474703a2f2f6578616d706c652e636f6d2f6d656d6f2f67656e65726963",
+                        MemoData: "72656e74"
+                    }
+                },
+                {
+                    Memo: {
+                        MemoFormat: "687474703a2f2f6578616d706c652e636f6d2f6d656d6f2f67656e65726963",
+                        MemoData: "72656e74"
+                    }
+                },
+                {
+                    Memo: {
+                        MemoType: "72656e74"
+                    }
+                }
+            ],
+            Signers: [
+                {
+                    Account: "r....",
```

Comment:
Should we check the shape of `Account` fields? e.g. right length, starts with `r`? Or is that too much?

Reply:
If we wanted checks on the value of account fields, we should use `ripple-blinary-codec` to check if its a valid address.

Reply:
But it will get caught in serialization.

Reply:
Ah that's fair.